src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
+        Node<K, V> currentNode = table[index];
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+        } else {
+            if (currentNode.hash == hashInputNode && (currentNode.key == key
+                    || (currentNode.key != null && currentNode.key.equals(key)))) {
+                currentNode.value = value;
+            } else {
+                for (; ; ) {
+                    if (currentNode.hash == hashInputNode && (currentNode.key == key
+                            || (currentNode.key != null && currentNode.key.equals(key)))) {
+                        currentNode.value = value;
+                        break;
+                    }
+                    if (currentNode.next == null) {
+                        currentNode.next = node;
+                        size++;
+                        break;
+                    }
+                    currentNode = currentNode.next;
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int hashInputNode = (key == null) ? 0 : key.hashCode();
+        int index = Math.abs(hashInputNode) % table.length;
+        if (index < table.length) {
+            Node<K, V> currentNode = table[index];
+            while (true) {
+                if (currentNode == null) {
+                    return null;
+                }
+                if (currentNode.hash == hashInputNode && (currentNode.key == key
+                        || (currentNode.key != null && currentNode.key.equals(key)))) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] newTable = new Node[table.length * 2];
+        int newIndex;
+        for (Node<K, V> node : table) {
+            Node<K, V> currentNode = node;
+            while (currentNode != null) {
+                newIndex = Math.abs(currentNode.hash) % newTable.length;
+                Node<K, V> newCurrentNode = newTable[newIndex];
+                if (newTable[newIndex] == null) {
+                    newTable[newIndex] = new Node<>(currentNode.hash,
+                            currentNode.key, currentNode.value, null);
+                } else {
+                    for (; ; ) {
+                        if (newCurrentNode.next == null) {
+                            newCurrentNode.next = new Node<>(currentNode.hash,
+                                    currentNode.key, currentNode.value, null);
+                            break;
+                        }
+                        newCurrentNode = newCurrentNode.next;
+                    }
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        table = newTable;
```

Comment:
too complex logic, rewrite it following this algorithm:
1. Create a new variable that will be storing old table
2. Set a new array for an old table with increased capacity
3. Go through each bucket of a variable created in step 1 and call the put method that you implemented

Reply:
method optimized